packages/stateful/actions/components/AuthzExec.tsx

```diff
@@ -0,0 +1,300 @@
+    },
+    {
+      type: AuthzExecActionTypes.Redelegate,
+      name: t('title.restake'),
+    },
+    {
+      type: AuthzExecActionTypes.ClaimRewards,
+      name: t('title.claimRewards'),
+    },
+    {
+      type: AuthzExecActionTypes.Custom,
+      name: t('title.custom'),
+    },
+  ]
+}
+
+export const AuthzExecComponent: ActionComponent<AuthzExecOptions> = (
+  props
+) => {
+  const { t } = useTranslation()
+  const { fieldNamePrefix, onRemove, errors, isCreating } = props
+  const { control, register, watch } = useFormContext()
+
+  const authzExecActions = useAuthzExecActionTypes()
+
+  const authzExecActionType = watch(fieldNamePrefix + 'authzExecActionType')
+
+  return (
+    <ActionCard
+      Icon={LockWithKeyEmoji}
+      onRemove={onRemove}
+      title={t('title.authzExec')}
+    >
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel
+          name={t('form.authzExecActionType')}
+          tooltip={t('form.authzExecActionTypeTooltip')}
+        />
+        <SelectInput
+          containerClassName="grow"
+          defaultValue={authzExecActions[0].type}
+          disabled={!isCreating}
+          error={errors?.authzExecActionType}
+          fieldName={fieldNamePrefix + 'authzExecActionType'}
+          register={register}
+        >
+          {authzExecActions.map(({ name, type }, idx) => (
+            <option key={idx} value={type}>
+              {name}
+            </option>
+          ))}
+        </SelectInput>
+      </div>
+
+      {authzExecActionType === AuthzExecActionTypes.Delegate && (
+        <>
+          <div className="flex flex-col items-stretch gap-1">
+            <InputLabel name={t('form.delegatorAddress')} />
+            <TextInput
+              disabled={!isCreating}
+              error={errors?.delegate?.deleatorAddress}
+              fieldName={fieldNamePrefix + 'delegate.delegatorAddress'}
+              placeholder="juno..."
+              register={register}
+              validation={[(v: string) => validateAddress(v)]}
+            />
+            <InputErrorMessage error={errors?.delegate?.delegatorAddress} />
+          </div>
+          <div className="flex flex-col items-stretch gap-1">
+            <InputLabel
+              name={t('form.validatorAddress')}
+              tooltip={t('form.validatorAddressTooltip')}
+            />
+            <TextInput
+              disabled={!isCreating}
+              error={errors?.delegate?.validatorAddress}
+              fieldName={fieldNamePrefix + 'delegate.validatorAddress'}
+              placeholder="junovaloper..."
+              register={register}
+              validation={[(v: string) => validateValidatorAddress(v)]}
+            />
+            <InputErrorMessage error={errors?.delegate?.validatorAddress} />
+          </div>
+          <div className="flex flex-col items-stretch gap-1">
+            <InputLabel name={t('title.chooseTokenAmount')} />
+            <TextInput
```

Comment:
It'd be better to use a `NumberInput` here to improve accessibility and keep consistent with other amount entries; setting `onMinus` and `onPlus` on a `NumberInput` make it look real cool too with the buttons. It would also be nice to set `unit` to `"$" + nativeTokenLabel(NATIVE_DENOM)` so that it's clearer what asset is being asked for.

Reply:
PERFECT :heart: